### ISSUE TITLE
Should I use face.embedding or face.normed_embedding for further linear transformation?

### ISSUE THREAD

**Issue by off99555** [deepinsight/insightface#2256]
1. Let's say I want to use the embedding to train a classification model or connect it to other models, should I normalize the embedding before I pass it to the next layer?
2. Is there any tutorials/articles/etc that talk about normalized embeddings vs unnormalized embeddings so that I know the intuition behind how to use them properly?
3. When training the ArcFace model, was the model trained to minimize loss on unnormalized embedding or normalized embedding?

**Comment by nttstar** [comment]
Generally, you should use the embedding after normalization.

**Comment by off99555** [comment]
OK. That answered my question. Thanks!

**Comment by off99555** [comment]
Actually I just tinkered about it more and it depends on the application. `face.embedding` returns a standardized embedding (std ~= 1 and mean ~= 0) so it's similar to Z score and this is useful for some applications that expect standardized values e.g. neural networks that work with `LayerNorm` from pytorch. These neural networks typically expect inputs to be standardized.

`face.normed_embedding` returns L2 normalized embedding which is also useful in some applications that expect the length of the embedding to be 1. Maybe some applications that expect the embeddings to lie on the surface of the unit sphere.
